capstone/src/main/java/com/google/step/search/ElasticsearchClient.java

```diff
@@ -79,14 +79,26 @@ public void updateRestaurantHeader(RestaurantHeader restaurantHeader) throws IOE
     request.execute();
   }
 
+  /**
+   * Queries search server for {@link RestaurantHeader} objects that match {@code query} by either
+   * "name" or "cuisine" field, if the query is not empty. If the query is empty, returns a random
+   * assortment of restaurant headers.
+   * @param query valid query string, or empty string
+   * @return list of {@link RestaurantHeader} objects sorted by descending relevance score
+   * @throws IOException thrown if request cannot be made or executed properly
+   */
+  public List<RestaurantHeader> searchRestaurants(String query) throws IOException {
+    return query.isEmpty() ? getRandomRestaurants() : queryRestaurantHeaders(query);
+  }
+
   /**
    * Queries search server for {@link RestaurantHeader} fields that match {@code query} by either
    * "name" or "cuisine" field.
    * @param query valid query string
    * @return list of {@link RestaurantHeader} objects sorted by descending relevance score
    * @throws IOException if request cannot be made or executed properly
    */
-  public List<RestaurantHeader> queryRestaurantHeaders(String query) throws IOException {
+  private List<RestaurantHeader> queryRestaurantHeaders(String query) throws IOException {
     List<String> requestPath = Arrays.asList("", RESTAURANTS, "_search");
 
     String requestBody = new JSONObject()
```

Comment:
What would actually happen if you just passed an empty string into this function as-is? Would it break? It would seem to me that elasticsearch should be able to match every record to an empty string, making getRandomRestaurants redundant.

Reply:
When Elasticsearch is queried with an empty string, it has no matches by default. I had to test this functionality on my own before writing the code for the client.

Reply:
I would still consider just having it check for empty string and structure the internal query as-is, unless there's some semantic you're trying to achieve with separating them out. In any case, we're splitting hairs so I don't want to delay things.

Reply:
I guess I separated them because the queries are structured in quite different ways. Perhaps I should create a query builder method of sorts that does this separation instead? I just feel like a single method can quickly become cluttered with the various ways there are to structure queries. If I were to have four or five different query types, possibly constructing all queries in a single method can quickly become cluttered.